Library/Homebrew/test/cmd/list_spec.rb

```diff
@@ -2,8 +2,24 @@
 
 require "cmd/list"
 require "cmd/shared_examples/args_parse"
+require "tab"
 
 RSpec.describe Homebrew::Cmd::List do
+  def setup_installation(formula_name, installed_on_request:)
+    setup_test_formula(formula_name)
+
+    keg_dir = HOMEBREW_CELLAR/formula_name/"1.0"
+    keg_dir.mkpath
+
+    tab = Tab.new(
+      "installed_on_request" => installed_on_request,
+      "tabfile"              => keg_dir/Tab::FILENAME,
+    )
+    tab.write
```

Comment:
Tangential: This is starting to crop up in the test code a bit more and I'm wondering if it'd be better to just add an option to `setup_test_formula` that facilitates that like `with_tab: { "installed_on_request" => true }`.

CC: @reitermarkus because we've talked about this a bit before

Reply:
@apainintheneck good idea 👍🏻